Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp

```diff
@@ -104,183 +98,137 @@ uint16_t  *plTextFont::IInitFontTexture()
-
-            tBits++;
-        }
-    }
+    SelectObject(hDC, hFont);
+    
+    DWORD fontDataSize = GetFontData( hDC, 0, 0, 0, 0 );
+    void* fontData = std::malloc( fontDataSize );
+    GetFontData(hDC, 0, 0, fontData, fontDataSize);
+    ftError = FT_New_Memory_Face(library, (FT_Byte *) fontData, fontDataSize, 0, &face);
+    FT_UInt freeTypeResolution = GetDeviceCaps(hDC, LOGPIXELSY);
 
-    // Cleanup and return
-    DeleteObject( hBitmap );
     DeleteDC( hDC );
     DeleteObject( hFont );
 
-    return data;
 #elif defined(HS_BUILD_FOR_APPLE)
-    int     nHeight, x, y, c;
-    char    myChar[ 2 ] = "x";
-    uint16_t  *tBits;
-
-    uint32_t       *bitmapBits;
-    uint32_t        bAlpha;
-
     
-    // Figure out our texture size
-    if( fSize * 2 > 40 )
-        fTextureWidth = fTextureHeight = 1024;
-    else if( fSize * 2 > 20 )
-        fTextureWidth = fTextureHeight = 512;
-    else
-        fTextureWidth = fTextureHeight = 256;
-
-
-    // Create a new DC and bitmap that we can draw characters to
-    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericRGB);
-    CGContextRef bitmapContext = CGBitmapContextCreate(nullptr,
-                                                fTextureWidth,
-                                                fTextureHeight,
-                                                8,
-                                                4 * fTextureWidth,
-                                                colorSpace,
-                                                kCGImageAlphaPremultipliedLast);
-    CFRelease(colorSpace);
-    CGContextSetTextMatrix(bitmapContext, CGAffineTransformMakeScale(1, -1));
-    CGContextTranslateCTM(bitmapContext, 0, fTextureHeight);
-    CGContextScaleCTM(bitmapContext, 1, -1);
-    CGContextScaleCTM(bitmapContext, 2.0, 2.0);
+    CFStringRef fontName = CFStringCreateWithCString(nullptr, fFace, kCFStringEncodingUTF8);
+    CTFontDescriptorRef fontDescriptor = CTFontDescriptorCreateWithNameAndSize( fontName, 0.0f );
+    CTFontDescriptorRef fulfilledFontDescriptor = CTFontDescriptorCreateMatchingFontDescriptor(fontDescriptor, nullptr);
+    hsAssert(fulfilledFontDescriptor != nullptr, "Cannot create Mac font");
+    CFRelease(fontName);
+    CFRelease(fontDescriptor);
+    
+    CFURLRef fontURL = (CFURLRef) CTFontDescriptorCopyAttribute(fulfilledFontDescriptor, kCTFontURLAttribute);
+    CFStringRef fileSystemPath = CFURLCopyFileSystemPath(fontURL, kCFURLPOSIXPathStyle);
+    char cPath[PATH_MAX];
+    CFStringGetCString(fileSystemPath, cPath, PATH_MAX, kCFStringEncodingUTF8);
+    
+    ftError = FT_New_Face(library, cPath, 0, &face);
     
-    bitmapBits = (uint32_t *)CGBitmapContextGetData(bitmapContext);
+    CFRelease(fulfilledFontDescriptor);
+    CFRelease(fontURL);
+    CFRelease(fileSystemPath);
+    
+    FT_UInt freeTypeResolution = 192;
+#else
+    FT_Done_FreeType(library);
+    FT_UInt freeTypeResolution = 0;
+    return nullptr;
+#endif
```

Comment:
I've put both Mac and Windows onto FreeType. The only thing platform specific is the TTF lookup. There isn't a huge amount of code - I think we can do this without a third party library. Hoping it's sufficient for @dpogue to get fonts working.

Reply:
I am looking into `fontconfig` on Linux as I type this 😉

Reply:
`fontconfig` works on all three platforms, right? It would be nice to use one library instead of having to maintain all of this going forward.

Reply:
I don't think fontconfig is something we'd want to use outside of Linux (and if I can avoid using it there, I'd like to 😛)

Reply:
I too would like to avoid using Fontconfig - the macOS code here should be pretty stable. Looking at Windows - the DirectWrite stuff might be even more trivial (mostly because it avoids the CFStringRef stuff going on with macOS.)